src/wp-includes/block-supports/colors.php

```diff
@@ -63,10 +63,10 @@ function wp_register_colors_support( $block_type ) {
  * @return array Colors CSS classes and inline styles.
  */
 function wp_apply_colors_support( $block_type, $block_attributes ) {
-	$color_support                 = _wp_array_get( $block_type->supports, array( '__experimentalColor' ), false );
+	$color_support                 = _wp_array_get( $block_type->supports, array( 'color' ), false );
 	$has_text_colors_support       = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'text' ), true ) );
 	$has_background_colors_support = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'background' ), true ) );
-	$has_link_colors_support       = _wp_array_get( $color_support, array( 'linkColor' ), false );
+	$has_link_colors_support       = _wp_array_get( $color_support, array( 'link' ), false );
```

Comment:
This change is unrelated to the issue reported but I noticed it was using the wrong key name as well.

Reply:
I guess this never worked on Core :). So it's a nice bug/feature for WP 5.7 :P

Reply:
I thought it'll save us future headaches!

PS for the passer-by: link color still won't be enabled in WordPress core because theme support is not ported.